scripts/build-dists-tarball.sh

```diff
@@ -0,0 +1,173 @@
+    osversion=$VERSION_ID
+  elif type lsb_release >/dev/null 2>&1; then
+    # linuxbase.org
+    osname=$(lsb_release -si)
+    osversion=$(lsb_release -sr)
+  elif [ -f /etc/lsb-release ]; then
+    # For some versions of Debian/Ubuntu without lsb_release command
+    . /etc/lsb-release
+    osname=$DISTRIB_ID
+    osversion=$DISTRIB_RELEASE
+  elif [ -f /etc/debian_version ]; then
+    # Older Debian/Ubuntu/etc.
+    osname=Debian
+    osversion=$(cat /etc/debian_version)
+  elif [ -f /etc/SuSe-release ]; then
+    # Older SuSE/etc.
+    echo "Suse?"
+  elif [ -f /etc/redhat-release ]; then
+    # Older Red Hat, CentOS, etc.
+    echo "RedHat?"
+  else
+    # Fall back to uname, e.g. "Linux <version>", also works for BSD, etc.
+    osname=$(uname -s)
+    osversion=$(uname -r)
+  fi
+
+  case $(uname -m) in
+    x86_64)
+      osarch="x64"  # or AMD64 or Intel64 or whatever
+      ;;
+    i*86)
+      osarch="x86"  # or IA32 or Intel32 or whatever
+      ;;
+    *)
+      # leave ARCH as-is
+      ;;
+  esac
+
+elif [ "$(expr substr $(uname -s) 1 10)" == "MINGW32_NT" ]; then
+  # Do something under 32 bits Windows NT platform
+  echo "32 bits Windows NT platform"
+elif [ "$(expr substr $(uname -s) 1 10)" == "MINGW64_NT" ]; then
+  # Do something under 64 bits Windows NT platform
+  echo "64 bits Windows NT platform"
+fi
+
+distFullName="$distName-$osname-$osversion-$osarch"
+echo $distFullName
+
+# Just a basic clean check
+if [ -n "$(git status --porcelain)" ]; then
+  echo "There are changes, please clean up before re-running $0";
+#  exit 1
+else
+  echo "No changes";
+fi
+
+git fetch --all --tags
+
+if [ "$1" == "latest-tag" ]; then
+  gitTagVersion=$(git describe --tags `git rev-list --tags --max-count=1`)
+  git checkout tags/$gitTagVersion -B $gitTagVersion-build
+else
+  gitTagVersion="${1}"
+fi
+
+echo $gitTagVersion
+
+git rev-parse --symbolic-full-name --abbrev-ref HEAD
+
+#if [ "git branch --list ${gitTagVersion-build}" ]; then
+#  git checkout tags/$gitTagVersion
+#else
+#  git checkout tags/$gitTagVersion -b $gitTagVersion-build
+#fi
+
+#git checkout tags/$gitTagVersion -b $gitTagVersion-build
+# As we will not push anything up using git, we don't need to track the branch?
+#git branch --set-upstream-to=origin/$gitTagVersion $gitTagVersion-build
+#git pull
+
+shaSumVal="256"
+#hashFile="$distFullName-$gitTagVersion.sha"
+#hashFile+="$shaSumVal"
+#hashFile+="sum"
+
+hashFile="$distFullName-$gitTagVersion.sha${shaSumVal}sum"
+archiveFile="$distFullName-$gitTagVersion.zip"
+echo $hashFile
+
+distDir=$(mktemp -d)
+if [ -d $distDir ]; then
+  echo "Temporary directory $distDir exists"
+else
+  echo "Temporary directory $distDir does not exist"
+  exit 2
+fi
+
+mkdir $distDir/dist
+
+cargo build --release
+
+COPY_FILES=(
+  "target/release/tari_base_node"
+  "config.toml"
+  "config/tari_config_sample.toml"
+  "log4rs.yml"
+  "common/logging/log4rs-sample.yml"
+  "applications/tari_base_node/README.md"
+  "install.sh"
+)
```

Comment:
You're missing a few files:

* scripts/install_tor.sh
* applications/tari_base_node/install-osx.sh
* applications/tari_base_node/start_tor.sh

Reply:
I am thinking that having each platform have their own folder for easier include and pointing users at in the repo, would make maintenance better.  
  
All three scripts are very OSX focused, but seem a little spread all over the repo.

Reply:
Files in `scripts` are aimed at developers and are "meta" tools; i.e. they're used to help with the application / source code. So something like this script (`build-dists-tarball.sh`) belongs there.

Scripts that are application specific (`install_tor`, the base node install script) are kept with the specific application's source code.

The config files are platform-agnostic and are used by multiple applications, hence they're in the common crate.

By that measure, `install_tor` is really in the wrong place. The others are fine.

You're spot-on about the OSX-specific nature of some of the scripts; let's build towards multi-platform support; it doesn't all have to go in this PR.

Reply:
Can you elaborate on which files would be in separate folders?
Like 
* win/install.bat
* osx/install_sh

that sort of thing?

Reply:
I would move
 - applications/tari_base_node/install-osx.sh -> ``osx/install.sh```
 - applications/tari_base_node/start_tor.sh -> ```osx/start_tor.sh```
 
and any other files or scripts that are ```osx``` or ```ubuntu``` or other platform specific.  
 
Then the dist build script can just include all the files in the platform folder into the archive.  
  
Don't think we will be able to do Win32/64 just yet, but I have not looked at cross-compiling in a long time.

Reply:
Yup, we can do that